libs/vkd3d/command.c

```diff
@@ -5444,6 +5446,7 @@ static void STDMETHODCALLTYPE d3d12_command_list_DrawInstanced(d3d12_command_lis
             return;
     }
 
+    d3d12_command_list_end_batch(list);
```

Comment:
This should be placed in begin_render_pass() I think. begin_render_pass is always done before a draw call.